### PR TITLE
Move package metadata to setup.cfg

### PR DESCRIPTION
**Pull request by jpy-git**
### Brief summary of the change made

This PR is a straight swap from `setup.py` to declarative metadata  in `setup.cfg`. Simplifies the setup file and makes it easier for tools to extract the package metadata (e.g. extracting requirements for use in the docker build #1945).

### Are there any other side effects of this change that we should be aware of?
No, exactly the same but uses declarative metadata.

### Pull Request checklist
- [X] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `python test/generate_parse_fixture_yml.py` or by running `tox` locally).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -1,3 +1 @@
 include README.md LICENSE.md CHANGELOG.md
-include src/sqlfluff/config.ini
-include src/sqlfluff/core/default_config.cfg
```
Comment (jpy-git) [inline MANIFEST.in:3]: These are handled in setup.cfg now via `[options.package_data]`. License is also added to `.whl` files via `license_files` metadata. `Manifest.in` is now only required for additional repo root level files we want included in the source dist.

File: setup.py [hunk 2]
```diff
@@ -1,134 +1,14 @@
...
-
-from setuptools import find_packages, setup
-
+from setuptools import setup
 
 # Get the global config info as currently stated
 # (we use the config file to avoid actually loading any python here)
 config = configparser.ConfigParser()
 config.read(["src/sqlfluff/config.ini"])
 version = config.get("sqlfluff", "version")
```
Comment (jpy-git) [inline setup.py:10]: Due to custom nature of the version info in sqlfluff we keep the version logic in setup.py.

A better practice for single source of version info is to set the version entry in setup.cfg and then reference in src/sqlfluff/__init__.py:
```python
import importlib.metadata
__version__ = importlib.metadata.version('sqlfluff')
```
however, I believe that config.ini is referenced in a couple locations so it makes sense to keep the version logic as is for now and I'll investigate the above approach in a subsequent PR.

**Review by jpy-git — COMMENTED** [review]
@barrywhart could you take a look at this when you get a chance please? Hoping to build off of it in the docker PR #1945, Thanks! 😄

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    snowflake
+    spark3
+    sqlite
+    teradata
+    tsql
+    dbt
+
+[options]
+package_dir =
+    =src
```
Comment (barrywhart) [inline setup.cfg:59]: Is it correct, this setting beginning with an `=`? (Looks odd.)

File: setup.cfg [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+install_requires =
+    # Used for finding os-specific application config dirs
+    appdirs
+    # Cached property for performance gains
+    cached-property
+    # To get the encoding of files.
+    chardet
+    click>=7.1
+    colorama>=0.3
+    configparser
```
Comment (barrywhart) [inline setup.cfg:71]: This module is built into Python 3, apparently. Can it be removed from here?

https://docs.python.org/3/library/configparser.html

**Review by barrywhart — COMMENTED** [review]
Looks good overall! A couple of small questions or suggestions.

File: setup.cfg (hunk 3 above)
Comment (jpy-git) [reply setup.cfg:59]: @barrywhart it is correct, it's the equivalent of package data setting in previous setup.py 👍 
image

I've tested building and it works as expected 

see ref here:
image

File: setup.cfg (hunk 4 above)
Comment (jpy-git) [reply setup.cfg:71]: let me give it a go

Comment (jpy-git) [reply setup.cfg]: @barrywhart I removed it and did a fresh reinstall of the environment and the unit tests all look good so I think we should be fine with removing it

**Review by barrywhart — APPROVED** [review]

**Pull request merged**